app/controllers/mixins/breadcrumbs_mixin.rb

```diff
@@ -196,5 +201,22 @@ def breadcrumbs_options
     def x_node_right_cell
       @sb[@sb[:active_accord]].presence || x_node
     end
+
+    def breadcrumb_get_session
+      @x_node_text = session[:x_node_text]
+    end
+
+    def breadcrumb_set_session
+      session[:x_node_text] = @x_node_text
+    end
+
+    def x_node_text
+      @x_node_text[x_active_tree] if @x_node_text
```

Comment:
You can get rid of the `if` if you realize that the `[]` call on the `Hash` is just a regular method:
```suggestion
      @x_node_text.try(:[], x_active_tree)
```